codegen/src/test/kotlin/software/amazon/smithy/rust/codegen/smithy/RuntimeTypesTest.kt

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.rust.codegen.smithy
+
+import io.kotest.matchers.comparables.shouldBeEqualComparingTo
+import io.kotest.matchers.equality.shouldBeEqualToComparingFields
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.MethodSource
+import software.amazon.smithy.model.node.Node
+import software.amazon.smithy.rust.codegen.rustlang.CratesIo
+import software.amazon.smithy.rust.codegen.rustlang.DependencyLocation
+import software.amazon.smithy.rust.codegen.rustlang.Local
+
+class RuntimeTypesTest {
+    @ParameterizedTest
+    @MethodSource("runtimeConfigProvider")
+    fun `succeeded to parse runtime config`(
+        runtimeConfig: String,
+        expectedCrateLocation: RuntimeCrateLocation,
+    ) {
+        val node = Node.parse(runtimeConfig)
+        val cfg = RuntimeConfig.fromNode(node.asObjectNode())
+        cfg.runtimeCrateLocation shouldBeEqualToComparingFields expectedCrateLocation
+    }
+
+    @ParameterizedTest
+    @MethodSource("runtimeCrateLocationProvider")
+    fun `runtimeCrateLocation provides dependency location`(
+        path: String?,
+        versions: CrateVersionMap,
+        crateName: String?,
+        expectedDependencyLocation: DependencyLocation,
+    ) {
+        val crateLoc = RuntimeCrateLocation(path, versions)
+        val depLoc = crateLoc.crateLocation(crateName)
+        depLoc.toString() shouldBeEqualComparingTo expectedDependencyLocation.toString()
+    }
+
+    companion object {
+        @JvmStatic
+        private val defaultVersion = defaultRuntimeCrateVersion()
+
+        @JvmStatic
+        fun runtimeConfigProvider() = listOf(
+            Arguments.of(
+                """{}""",
+                RuntimeCrateLocation(null, CrateVersionMap(mapOf())),
+            ),
+            Arguments.of(
+                """{
+                "relativePath": "/path"
+                }""",
```

Comment:
```suggestion
                """
                {
                    "relativePath": "/path"
                }
                """,
```

nit: if a `"""` string is multiline, I think the style we're following is to put the `"""` in its own line.